BepInEx.Core/Contract/Attributes.cs

```diff
@@ -234,22 +206,22 @@ internal static IEnumerable<BepInIncompatibility> FromCecilType(TypeDefinition t
 }
 
 /// <summary>
-///     This attribute specifies which processes this plugin should be run for. Not specifying this attribute will load the
+///     This attribute specifies which processes this <see cref="Plugin"/> should be run for. Not specifying this attribute will load the
 ///     plugin under every process.
 /// </summary>
 [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
 public class BepInProcess : Attribute
 {
-    /// <param name="ProcessName">The name of the process that this plugin will run under.</param>
-    public BepInProcess(string ProcessName)
+    /// <param name="processName">The name of the process that this plugin will run under.</param>
+    public BepInProcess(string processName)
     {
-        this.ProcessName = ProcessName;
+        ProcessName = processName;
     }
 
     /// <summary>
     ///     The name of the process that this plugin will run under.
     /// </summary>
-    public string ProcessName { get; protected set; }
+    public string ProcessName { get; }
 
     internal static List<BepInProcess> FromCecilType(TypeDefinition td)
```

Comment:
Since every attribute needs a FromCecilType maybe have all attributes inherit from an abstract class with this method in it as abstract?

Reply:
Overridable methods cannot be static. We also can't have a static member in an interface either so this is why we have to define this once per type